packages/ts/frontend/src/Authentication.ts

```diff
@@ -59,12 +61,73 @@ export interface LoginResult {
   defaultUrl?: string;
 }
 
+export type SuccessCallback = () => Promise<void> | void;
+
+export type NavigateFunction = (path: string) => void;
+
 export interface LoginOptions {
+  /**
+   * The URL for login request, defaults to `/login`.
+   */
   loginProcessingUrl?: string;
```

Comment:
Maybe, we should support `URL` here or change the naming? It is a bit confusing that `...Url` does not support real `URL`.